libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStoreInspectorViewController.m

```diff
@@ -88,14 +88,28 @@ + (SFSmartStoreInspectorViewController *) sharedInstance
 
 #pragma mark - Present / dimiss
 
-+ (void) present
++ (void) present:(UIViewController*)currentViewController
 {
-    [[SFRootViewManager sharedManager] pushViewController:[SFSmartStoreInspectorViewController sharedInstance]];
+    if (![NSThread isMainThread]) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [SFSmartStoreInspectorViewController present:currentViewController];
+        });
+        return;
+    }
+
+    [currentViewController presentViewController:[SFSmartStoreInspectorViewController sharedInstance] animated:NO completion:nil];
```

Comment:
The reason is that if you have typed a complex query then went back into the app and then back to the inspector it would still be there. 
We could also leave it to the devs to keep the instance around if they need that behavior.
So present/dismiss could become instance methods instead also, the constructor could take a store to be able to inspect the global stores or other stores.